packages/contract/src/contracts/contract.test.ts

```diff
@@ -36,13 +37,36 @@ describe('Contract', () => {
     expect(contract.wallet).toBe(null);
   });
 
+  test('Contract instance with provider can get balance', async () => {
+    const provider = new Provider('http://localhost:4000/graphql');
+    const contract = new Contract(CONTRACT_ID, ABI, provider);
+    await expect(contract.getBalance([1])).resolves.not.toThrow(
+      'Contract instance has no provider.'
+    );
+  });
+
   test('Create contract instance with wallet', async () => {
     const wallet = Wallet.generate();
     const contract = new Contract(CONTRACT_ID, ABI, wallet);
     expect(contract.provider).toBe(wallet.provider);
     expect(contract.wallet).toBe(wallet);
   });
 
+  test('Contract instance with wallet can get balance', async () => {
+    const wallet = Wallet.generate();
+    const contract = new Contract(CONTRACT_ID, ABI, wallet);
+    await expect(contract.getBalance([1])).resolves.not.toThrow(
+      'Contract instance has no provider.'
+    );
+  });
+
+  test('Contract instance can multi call functions', async () => {
+    const wallet = Wallet.generate();
+    const contract = new Contract(CONTRACT_ID, ABI, wallet);
+    const calls = [contract.functions.foo(123)];
+    expect(contract.multiCall(calls)).toBeInstanceOf(MultiCallInvocationScope);
+  });
```

Comment:
Thankyou for your input @camsjams 😃 Yes this is a very basic validation check for module level coverage, it would be good to understand the test plan for the project, not sure if that is documented anywhere? In some places, especially within fuel-gauge, it seems to cover more feature level testing of a range of packages/functionality. And then some packages do seem to utilise more unit level testing. 

I am quite pro utilising both, rather than just feature, especially in the case of SDKs where modules themselves could be further extrapolated and packages then become untested in isolation.

Reply:
@danielbate You got it right; the `fuel-gauge` package currently groups all our e2e tests interacting with the Fuel-Core node, while most unit tests are kept side-by-side with their source counterpart.